common/storage.ts

```diff
@@ -0,0 +1,104 @@
+import * as Constants from 'config/constants';
+import * as fs from 'node:fs';
+import { Storage } from '@google-cloud/storage';
+import { Stream } from "stream";
+import { arrayBuffer } from 'node:stream/consumers';
+import { createGzip, createGunzip } from 'zlib';
+import { pipeline } from 'node:stream/promises';
+
+import type { Bucket } from '@google-cloud/storage';
+
+export interface StorageAccessor {
+  readBytes(path: string) : Promise<ArrayBuffer>;
+  writeBytes(path: string, data: string) : Promise<unknown>;
+
+  // Returns uncompressed bytes from an path using lzma.
+  readBytesLzma(path: string) : Promise<ArrayBuffer>;
+  writeBytesLzma(path: string, data: string) : Promise<unknown>;
+
+  readBytesGzip(path: string) : Promise<ArrayBuffer>;
+  writeBytesGzip(path: string, data: string) : Promise<unknown>;
+}
+
+type CloudStorageAccessorOptions = {
```

Comment:
If this is only used in the constructor, I'd probably inline and destructure it, like the props in a React component.

```
constructor({
  keyfile?: string,
  makeLzmaDecompressor?: (() => object),
  makeLzmaCompressor?: (() => object),
} = {}) {
...
}
```

Reply:
done.

Reply:
erp...apparently the ?: doesn't work in the desctrucure?

Reply:
Weird